docs/content/modules/carts-and-checkout/backend/add-payment-provider.md

```diff
@@ -165,41 +174,46 @@ class MyPaymentService extends AbstractPaymentService {
 }
 ```
 
-### createPayment
+### PaymentProcessorError
+
+Before diving into the methods you'll need to implement, you'll notice that part of the expected return signature of these method includes `PaymentProcessorError`. This is an interface of the following definition:
+
+```ts
+interface PaymentProcessorError {
+  error: string
+  code?: string
+  detail?: any
+}
+```
+
+While implementing the following methods, if you need to inform the Medusa core that an error occurred at a certain stage, return an object having the attributes defined in the `PaymentProcessorError` interface.
```

Comment:
suggestion: We can maybe show an example from stripe or Paypal just to give a better idea to the user? Most users are a bit lost and this step is important to gracefully handle the errors